ktor-client/ktor-client-core/common/src/io/ktor/client/features/HttpPlainText.kt

```diff
@@ -13,51 +13,165 @@ import kotlinx.io.core.*
+
         /**
          * Default [Charset] to use.
          */
+        @Deprecated(
+            "Use [accept] method instead.",
+            replaceWith = ReplaceWith("accept()"),
+            level = DeprecationLevel.ERROR
+        )
         var defaultCharset: Charset = Charsets.UTF_8
-
-        internal fun build(): HttpPlainText = HttpPlainText(defaultCharset)
     }
 
     @Suppress("KDocMissingDocumentation")
     companion object Feature : HttpClientFeature<Config, HttpPlainText> {
         override val key = AttributeKey<HttpPlainText>("HttpPlainText")
 
-        override fun prepare(block: Config.() -> Unit): HttpPlainText = Config().apply(block).build()
+        override fun prepare(block: Config.() -> Unit): HttpPlainText {
+            val config = Config().apply(block)
+
+            with(config) {
+                return HttpPlainText(
+                    acceptCharsets, charsetQuality,
+                    sendCharset, responseCharsetFallback
+                )
+            }
+        }
 
         override fun install(feature: HttpPlainText, scope: HttpClient) {
             scope.requestPipeline.intercept(HttpRequestPipeline.Render) { content ->
+                feature.addCharsetHeaders(context)
+
                 if (content !is String) return@intercept
-                val contentType = ContentType.Text.Plain.withCharset(feature.defaultCharset)
-                proceedWith(TextContent(content, contentType))
+
+                val contentType = context.contentType()
+                if (contentType != null) return@intercept
+
+                proceedWith(feature.wrapContent(content))
             }
 
             scope.responsePipeline.intercept(HttpResponsePipeline.Parse) { (info, body) ->
                 if (info.type != String::class || body !is ByteReadChannel) return@intercept
-
                 val content = feature.read(context, body.readRemaining())
                 proceedWith(HttpResponseContainer(info, content))
             }
         }
     }
+
+    private fun wrapContent(content: String): Any {
+        return TextContent(content, ContentType.Text.Plain.withCharset(requestCharset))
+    }
+
+    internal fun read(call: HttpClientCall, body: Input): String {
+        val actualCharset = call.response.charset() ?: responseCharsetFallback
+        return body.readText(charset = actualCharset)
+    }
+
+    internal fun addCharsetHeaders(context: HttpRequestBuilder) {
+        if (context.headers[HttpHeaders.AcceptCharset] != null) return
+        context.headers[HttpHeaders.AcceptCharset] = acceptCharsetHeader
+    }
+
+    /**
+     * Deprecated
+     */
+    @Deprecated(
+        "Use [Config.accept] method instead.",
+        replaceWith = ReplaceWith("accept()"),
+        level = DeprecationLevel.ERROR
+    )
+    var defaultCharset: Charset
+        get () = error("defaultCharset is deprecated")
+        set(_) = error("defaultCharset is deprecated")
```

Comment:
Well, at least setter could be implemented by invoking `accept` to be binary compatible.